syft/internal/fileresolver/unindexed_directory.go

```diff
@@ -235,10 +238,16 @@ func (u UnindexedDirectory) AllLocations() <-chan file.Location {
 				return nil
 			}
 			p = strings.TrimPrefix(p, "/")
-			out <- file.NewLocation(p)
-			return nil
+			for {
```

Comment:
I wasn't certain whether this loop was necessary; I'd welcome some feedback from reviewers.

Reply:
It doesn't seem necessary. Are you seeing functional differences when removing it?

Reply:
I'll remove it; I think I confused myself about the semantics of a single `select`. (Namely: a single select with no default will block until it has something to do, but once the context is cancelled it will let the goroutine return immediately.)

I'll remove it.